packages/context/src/binding-sorter.ts

```diff
@@ -0,0 +1,124 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/context
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {Binding} from './binding';
+
+/**
+ * Compare function to sort an array of bindings.
+ * It is used by `Array.prototype.sort()`.
+ *
+ * @example
+ * ```ts
+ * const compareByKey: BindingComparator = (a, b) => a.key.localeCompare(b.key);
+ * ```
+ */
+export interface BindingComparator {
+  /**
+   * Compare two bindings
+   * @param bindingA First binding
+   * @param bindingB Second binding
+   * @returns A number to determine order of bindingA and bindingB
+   * - 0 leaves bindingA and bindingB unchanged
+   * - <0 bindingA comes before bindingB
+   * - >0 bindingA comes after bindingB
+   */
+  (
+    bindingA: Readonly<Binding<unknown>>,
+    bindingB: Readonly<Binding<unknown>>,
+  ): number;
+}
+
+/**
+ * Creates a binding compare function to sort bindings by tagged group name.
+ *
+ * @remarks
+ * Two bindings are compared as follows:
+ *
+ * 1. Get values for the given tag as `group` for bindings, if the tag is not
+ * present, default `group` to `''`.
+ * 2. If both bindings have `group` value in `orderedGroups`, honor the order
+ * specified by `orderedGroups`.
+ * 3. If a binding's `group` does not exist in `orderedGroups`, it comes before
+ * the one with `group` exists in `orderedGroups`.
+ * 4. If both bindings have `group` value outside of `orderedGroups`, they are
+ * ordered by group names alphabetically and symbol values come before string
+ * values.
+ *
+ * @param groupTagName Name of the binding tag for group
+ * @param orderedGroups An array of group names as the predefined order
+ */
+export function compareBindingsByGroup(
+  groupTagName: string = 'group',
+  orderedGroups: (string | symbol)[] = [],
```

Comment:
"orderedGroups" feels weird when using a tag name different from "group". How about `orderedTagValues`?

Reply:
Switched to `compareBindingsByTag`.